src/WebJobs.Script.WebHost/Security/SecretManager.cs

```diff
@@ -56,33 +69,46 @@ protected virtual void Dispose(bool disposing)
             }
         }
 
-        public virtual HostSecrets GetHostSecrets()
+        public virtual HostSecretsInfo GetHostSecrets()
         {
             if (_hostSecrets == null)
             {
                 string secretFilePath = Path.Combine(_secretsPath, ScriptConstants.HostMetadataFileName);
+                var hostSecrets = new HostSecrets();
+
                 if (File.Exists(secretFilePath))
                 {
                     // load the secrets file
                     string secretsJson = File.ReadAllText(secretFilePath);
-                    _hostSecrets = JsonConvert.DeserializeObject<HostSecrets>(secretsJson);
+                    hostSecrets = ScriptSecretSerializer.DeserializeHostSecrets(secretsJson);
                 }
                 else
                 {
                     // initialize with new secrets and save it
-                    _hostSecrets = new HostSecrets
+                    hostSecrets = new HostSecrets
```

Comment:
So we'll have both HostSecrets and HostSecretsInfo classes? Can't we get rid of the old one to reduce confusion?